### ISSUE TITLE
Do not report MA0011/MA0076 for use of ToString() on custom types when "null" format is defined as culture invariant

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#896]
I propose not to consider (the parameterless) `ToString()` as culture variant if a **CultureInsensitiveType** is applied (on the assembly or type) with value **null** for the **format** argument.

For example:
```c#
using System;
using Meziantou.Analyzer.Annotations;

internal sealed class Program
{
    private static void Main(string[] args)
    {
        Foo foo = new Foo();

        Console.WriteLine(foo.ToString()); // MA0011 should not be reported
        Console.WriteLine($"{foo.ToString()}"); // MA0011 should not be reported
        Console.WriteLine($"{foo}"); // MA0076 should not be reported
    }
}

[CultureInsensitiveType(format: null)]
public sealed class Foo : IFormattable
{
    public string ToString(string format)
    {
        return string.Empty;
    }

    public string ToString(IFormatProvider formatProvider)
    {
        return string.Empty;
    }

    public string ToString(string format, IFormatProvider formatProvider)
    {
        return string.Empty;
    }

    public override string ToString()
    {
        return string.Empty;
    }
}
```

There may be better ways to deal with this though.

**Comment by meziantou** [comment]
What you want is a way to indicates that the default format (when not specified) is invariant. What about the following

```c#
[CultureInsensitiveType()] // Always invariant whatever the format

[CultureInsensitiveType(format: null)] // ToString(null)
[CultureInsensitiveType(format: "something")] // ToString("something")
[CultureInsensitiveType(isDefaultFormatCultureInsensitive: true)] // ToString()
```

**Comment by drieseng** [comment]
@meziantou, I'm ok with this approach. Thanks!
